stripe/src/main/java/com/stripe/android/view/CardNumberEditText.kt

```diff
@@ -246,5 +251,12 @@ class CardNumberEditText @JvmOverloads constructor(
                 else -> MAX_LENGTH_COMMON
             }
         }
+
+        @JvmSynthetic
+        internal fun createFormattedNumber(cardParts: Array<String?>): String {
+            return cardParts
+                .takeWhile { it != null }
```

Comment:
`filterNotNull` might be simpler https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/filter-not-null.html

Reply:
I wanted to preserve the initial logic, and I don't think using `filterNotNull` would be the exact same thing (e.g. think of the case of `arrayOf("4242", "4242", null, "4242")`)

Reply:
ah yeah that's different :) good point!